src/pages/AdminPages/AdminDashboard/index.jsx

```diff
@@ -1,32 +1,43 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { useSelector } from 'react-redux'
 import Authenticate from '../../../components/Authenticate'
 import PageNotFound from '../../PageNotFound'
-import { useHistory } from 'react-router-dom';
-
+import { useDispatch, shallowEqual } from 'react-redux';
+import { getTraders } from "../../../store/reducers/adminQuery";
+import { setIsLoading } from '../../../store/reducers/user';
+import { Link } from 'react-router-dom';
 
 const AdminDashboard = () => {
-    const history = useHistory()
+    const dispatch = useDispatch()
     const { role, isLoading } = useSelector(state => ({
         role : state.user.role,
-        isLoading : state.isLoading
-    }))
+        isLoading : state.user.isLoading,
+        alterUsers: state.adminQuery.alterUsers,
+        traders: state.adminQuery.traders,
+    }), shallowEqual)
+
+    useEffect(() => {
+        dispatch(setIsLoading());
+        dispatch(getTraders())
```

Comment:
move this dispatch call into the `AlterUsers` file. by doing this here you've finalized that we must have the traders array persist into any page we link from here... and remember I said, we cannot make any assumptions for what will happen on the dashboards for now... not until we are done with the inner functions.